src/Parser.js

```diff
@@ -506,6 +514,18 @@ Parser.prototype.parseFunction = function(pos, mode) {
     }
 };
 
+/**
+ * Call a function handler with a suitable context and arguments.
+ */
+Parser.prototype.callFunction = function(name, data, args, positions) {
```

Comment:
Instead of passing in "data" we could use "name" to do the lookup from the "functions" array.